### ISSUE TITLE
angular.json with comments / trailing commas

### ISSUE THREAD

**Issue by dtaylor84** [cyrilletuzi/vscode-angular-schematics#153]
# Checks before posting an issue

- [x] I have read all the extension's documentation (README and docs/TROUBLESHOOTING)
- [x] I launched `ng g component hello` in the *VS Code* Terminal inside my project, and it works
- [x] I looked at the extension's logs ("Angular Schematics" Output channel, second tab left to the Terminal), and tried to fix my project's configuration accordingly
- [x] I understand I am writing to a *human being*, doing this open source project on his *free* time

# Description of the issue

The Angular CLI supports (or at least works with) trailing commas and comments in angular.json. 

Unfortunately, the schematics extension cannot parse angular.json if these are present. It results in the slightly confusing notification 'No valid Angular config file found for the chosen workspace. Add a "angular.json" file in your project with `{ "version": 1 }`'.

I realise that comments and trailing commas are strictly invalid in JSON documents, but it would be nice to be consistent with Angular itself in accepting them.

# How to reproduce the issue

1. Add a trailing comma or // line comment or /* block comment */ to angular.json
2. Run (e.g.) ng lint (works)
3. Try to use the angular schematics extension (breaks)

**Comment by cyrilletuzi** [comment]
On a feature perspective, I am not opposed to this. On a performance and security perspective however, it would require to add a third-party dependency (while currently there are none at all, the extension is autonomous).

I'll do some tests to evaluate this feature request.

**Comment by cyrilletuzi** [comment]
After a quick investigation:
- [strip-json-comments](https://github.com/sindresorhus/strip-json-comments/) would be acceptable as it is a very small lib with no dependencies; but it only removes comments; but while comments can be considered as an useful feature, trailings commas are just about style so it's really not a priority to support;
- [json5](https://github.com/json5/json5) supports all JSON5 features, but had a dependency on minimist, which has a security issue recently; so using this lib is blocked on json5/json5#205

**Comment by cyrilletuzi** [comment]
Note: Angular CLI is using its [own parser](https://github.com/angular/angular-cli/blob/master/packages/angular_devkit/core/src/json/parser.ts).

**Comment by cyrilletuzi** [comment]
@dtaylor84 Support for JSON5 has been added in 4.1.0.

For info: v4 took me more than a *full-time* non-paid week to do. Yet I still do not have any **[sponsor](https://github.com/sponsors/cyrilletuzi)**. So please consider to help, or to ask your enterprise, which makes money with this extension, to help.

**Comment by satabol** [comment]
now is 2023. Comments are underlined with red colors. I just ignore this alerts. (
